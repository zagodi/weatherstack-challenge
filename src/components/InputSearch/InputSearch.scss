.input-search-container {
  display: flex;

  button {
    margin-left: 14px;
  }
}

@media (max-width: 600px) {
  .input-search-container {
    flex-direction: column;
  }

  .input-search-container button {
    width: 100%;
    margin-left: 0;
    margin-top: 10px;
  }
}
