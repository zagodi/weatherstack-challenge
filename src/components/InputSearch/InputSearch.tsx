import React from "react";

import { Button, Input } from "antd";

import "./InputSearch.scss";

type PROPS = {
  value?: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onClickSearch: () => void;
};

const InputSearch = ({ onChange, value, onClickSearch }: PROPS) => {
  const handleEnter = () => {
    console.log("OI");
    if (!!value) {
      console.log("KKK");
      onClickSearch();
    }
  };

  return (
    <div className="input-search-container">
      <Input
        value={value}
        onChange={onChange}
        onPressEnter={handleEnter}
        placeholder="Type the city name"
      />

      <Button type="primary" onClick={onClickSearch} disabled={!value}>
        Search
      </Button>
    </div>
  );
};

export default InputSearch;
