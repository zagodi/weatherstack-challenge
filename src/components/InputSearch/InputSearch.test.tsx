import React from "react";

import { render, screen, fireEvent } from "@testing-library/react";

import InputSearch from "./InputSearch";

describe("<InputSearch />", () => {
  const mockOnChange = jest.fn();
  const mockOnClickSearch = jest.fn();

  it("should render the input and button correctly", () => {
    render(
      <InputSearch
        value=""
        onChange={mockOnChange}
        onClickSearch={mockOnClickSearch}
      />
    );

    expect(
      screen.getByPlaceholderText("Type the city name")
    ).toBeInTheDocument();

    const button = screen.getByRole("button", { name: "Search" });
    expect(button).toBeInTheDocument();
    expect(button).toBeDisabled();
  });

  it("should call onChange when input value changes", () => {
    render(
      <InputSearch
        value=""
        onChange={mockOnChange}
        onClickSearch={mockOnClickSearch}
      />
    );

    const input = screen.getByPlaceholderText("Type the city name");

    fireEvent.change(input, { target: { value: "New York" } });

    expect(mockOnChange).toHaveBeenCalled();
  });

  it("should enable the button when input has value and call onClickSearch when button is clicked", () => {
    render(
      <InputSearch
        value="New York"
        onChange={mockOnChange}
        onClickSearch={mockOnClickSearch}
      />
    );

    const button = screen.getByRole("button", { name: "Search" });

    expect(button).toBeEnabled();

    fireEvent.click(button);

    expect(mockOnClickSearch).toHaveBeenCalled();
  });

  it("should call onClickSearch when enter is pressed", () => {
    render(
      <InputSearch
        value="New York"
        onChange={mockOnChange}
        onClickSearch={mockOnClickSearch}
      />
    );

    const input = screen.getByPlaceholderText("Type the city name");

    fireEvent.keyPress(input, { key: "Enter", code: "Enter", charCode: 13 });

    expect(mockOnClickSearch).toHaveBeenCalled();
  });

  it("should match the snapshot", () => {
    const { container } = render(
      <InputSearch
        value=""
        onChange={mockOnChange}
        onClickSearch={mockOnClickSearch}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
