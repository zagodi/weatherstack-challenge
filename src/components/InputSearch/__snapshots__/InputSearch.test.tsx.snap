// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSearch /> should match the snapshot 1`] = `
<div>
  <div
    class="input-search-container"
  >
    <input
      class="ant-input css-dev-only-do-not-override-1hpnbz2 ant-input-outlined"
      placeholder="Type the city name"
      type="text"
      value=""
    />
    <button
      class="ant-btn css-dev-only-do-not-override-1hpnbz2 ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
      disabled=""
      type="button"
    >
      <span>
        Search
      </span>
    </button>
  </div>
</div>
`;
